### PR TITLE
add driver implementation

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

File: protocol/drivermain.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
+package protocol
```
Comment (mcuadros) [inline protocol/drivermain.go]: maybe instead will be best `github.com/bblfsh/sdk/protocol/driver/cmd/`

Comment (mcuadros) [inline protocol/drivermain.go]: and the main in the driver, following the same pattern of sdk cmd

File: protocol/internal/testdriver/main.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package main
```
Comment (mcuadros) [inline protocol/internal/testdriver/main.go]: why internal?

Comment (smola) [inline protocol/internal/testdriver/main.go]: `testdriver` package is used only in tests, this `main.go` then can be used (with a simple replacement) in `etc/skeleton/driver/main.go.tpl`.

File: protocol/drivermain.go (hunk 1 above)
Comment (smola) [inline protocol/drivermain.go]: Ok. Note that this does not build a command itself, but it's used from driver's main.go.

**Comment by smola** [comment]
This PR depends on https://github.com/bblfsh/sdk/pull/24

**Pull request merged**